cpp/src_prims/cuda_utils.h

```diff
@@ -109,6 +109,35 @@ std::string arr2Str(const T *arr, int size, std::string name, cudaStream_t strea
 }
 
 
+template<typename T = size_t>
+bool verify_size(T size, int device) {
```

Comment:
I actually wanted to centralize this function because I'm not so sure it's going to stick around. The creation of this function predates the complications that we face querying for free memory in an RMM pool. I've decided to keep it around for now, since I don't believe we have officially adopted the RMM allocator across the codebase yet. It should be revisited very soon, though, or maybe just naturally throw an exception in the case where not enough memory could be allocated. 

Thoughts?